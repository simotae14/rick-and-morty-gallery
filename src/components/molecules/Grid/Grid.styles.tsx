import styled from 'styled-components';

export const DefaultGrid = styled.div`
  display: flex;
  flex-wrap: wrap;
  max-width: 1220px;
  width: 100%;
  overflow: auto;
  @media (max-width: 1220px) {
    justify-content: center;
    padding-bottom: 96px;
  }
`;
