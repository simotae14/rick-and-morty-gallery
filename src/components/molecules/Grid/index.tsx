import { DefaultGrid } from './Grid.styles';

/**
 * Component used to show the cards in a structured way
 */
function Grid({
  children,
}: {
  /**
   * It is the content wrapped inside the layout
   */
  children: React.ReactNode;
}) {
  return <DefaultGrid>{children}</DefaultGrid>;
}

export default Grid;
