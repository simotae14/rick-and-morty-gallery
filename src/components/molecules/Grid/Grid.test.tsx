/* eslint-disable import/no-extraneous-dependencies */
import { describe, it } from 'vitest';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';

import Grid from '.';

expect.extend(toHaveNoViolations);

describe('Components => molecules => Grid', () => {
  it('should render the Grid', () => {
    const { asFragment } = render(
      <Grid>
        <div>div1</div>
        <div>div2</div>
        <div>div3</div>
        <div>div4</div>
        <div>div5</div>
        <div>div6</div>
        <div>div7</div>
        <div>div8</div>
        <div>div9</div>
        <div>div10</div>
        <div>div11</div>
        <div>div12</div>
      </Grid>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not fail any accessibility tests', async () => {
    const { container } = render(
      <Grid>
        <div>div1</div>
        <div>div2</div>
        <div>div3</div>
        <div>div4</div>
        <div>div5</div>
        <div>div6</div>
        <div>div7</div>
        <div>div8</div>
        <div>div9</div>
        <div>div10</div>
        <div>div11</div>
        <div>div12</div>
      </Grid>
    );
    expect(await axe(container)).toHaveNoViolations();
  });
});
