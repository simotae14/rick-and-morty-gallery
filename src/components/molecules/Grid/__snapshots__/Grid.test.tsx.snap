// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components => molecules => Grid > should render the Grid 1`] = `
<DocumentFragment>
  <div
    class="sc-eLqIir hWmaC"
  >
    <div>
      div1
    </div>
    <div>
      div2
    </div>
    <div>
      div3
    </div>
    <div>
      div4
    </div>
    <div>
      div5
    </div>
    <div>
      div6
    </div>
    <div>
      div7
    </div>
    <div>
      div8
    </div>
    <div>
      div9
    </div>
    <div>
      div10
    </div>
    <div>
      div11
    </div>
    <div>
      div12
    </div>
  </div>
</DocumentFragment>
`;
